### PR TITLE
Break DB integration tests out into their own package

### PR DESCRIPTION
**Pull request by tcsc**
Making all of our integration tests run in entirely parallel requires
a large engineering effort to enforce test isolation and remove all race
conditions between tests.

A lower-effort alternative may be to split apart the various test suites
into their own Go packages, and test those packages in parallel, even if
the tests inside are still executed serially. Auditing the test suites
for races on system-level resources (e.g. files, ports) is much easier
than chasing down every possible race in the testing system.

This patch acts as a trial run, breaking a fairly well-defined and
self-contained test suite out into its own package. Note that the goal of
this change is not necessarily to shave minutes off the build (although
that would be nice), but to act as an illustration of how other, less
well-formed test suites might be broken apart.

See-Also: #12421
See-Also: #14408

### CODE REVIEW

**Comment by strideynet** [comment]
Motivation is sound and a move in the right direction of making our integration tests less painful. Strategy looks good. Will review the code here once this is opened fully.

**Comment by tcsc** [comment]
Looks like setting up the test DB servers is going to take a bit more effort than I anticipated. There are a few sequencing issues that make it hard to get the test servers running arbitrary ports while still getting the Teleport DB services to talk to them.

Oh, well . Maybe fresh eyes in the morning.

File: integration/helpers/helpers.go [hunk 1]
```diff
@@ -216,3 +213,47 @@ func MustCreateUserIdentityFile(t *testing.T, tc *helpers.TeleInstance, username
...
+			return nil
+		}
+
+		time.Sleep(500 * time.Millisecond)
+	}
+
+	return trace.BadParameter("proxy count on %v: %v (wanted %v)", clusterName, counts[clusterName], count)
+}
+
+func WaitForAuditEventTypeWithBackoff(t *testing.T, cli *auth.Server, startTime time.Time, eventType string) []apievents.AuditEvent {
```
Comment (strideynet) [inline integration/helpers/helpers.go:234]: Should this function be marked with `t.Helper()`

**Review by strideynet — APPROVED** [review]
Looks good. I had a few thoughts on the tests themselves but since it's pre-existing code moving, it's easier to let it go for now.

**Review by espadolini — APPROVED** [review]
The GCB run wasn't faster than other integration test runs that we've had - is that expected given the variance, or is there some other factor at play that still serializes the testing?

**Comment by tcsc** [comment]
> The GCB run wasn't faster than other integration test runs that we've had - is that expected given the variance, or is there some other factor at play that still serializes the testing?

I wasn't seriously expecting an immediately-obvious difference with this change. The 2 mins or so it takes the DB tests is well within the everyday variability of the integration tests, so hopefully the average time will drop a tiny bit, but I'm not expecting a night-and-day change just yet.

**Pull request merged**